eng/pipelines/templates/steps/build-and-test.yml

```diff
@@ -48,6 +51,8 @@ steps:
       jdkArchitectureOption: 'x64'
       publishJUnitResults: false
       goals: 'install'
+    env: 
+      AZURE_VERSION_OVERRIDE_TESTS: ${{ parameters.TestVersionSupport }}
```

Comment:
@lmolkova @alzimmermsft I think the parameter for `TestVersionSupport` is redundant and should be removed, as removing parameter bloat in the java templates is an ongoing effort. As it's being defined as a variable in the matrix configs anyway, you can instead set a default in `eng/pipelines/templates/variables/globals.yml` and reference `$(TestVersionSupport)` directly here to handle the null case.